specifyweb/frontend/js_src/lib/components/QueryBuilder/Formatter.tsx

```diff
@@ -173,13 +203,26 @@ function FormatSelect({
             className={customSelectElementBackground}
             disabled={handleChange === undefined}
             id={id('list')}
-            value={currentFormat}
-            onValueChange={handleChange}
+            value={compositeValue}
+            onValueChange={(value) => {
+              const parsed = parseFormatterValue(value);
+              if (parsed) {
+                setSelectedIndex(parsed.index);
+                if (handleChange) {
+                  handleChange(parsed.name);
+                }
+              } else {
+                setSelectedIndex(null);
+                if (handleChange) {
+                  handleChange(value);
+                }
+              }
```

Comment:
(optional) some shorthands if you want to make it cleaner
```suggestion
              const index = parsed?.index ?? null;
              const name = parsed?.name ?? value;
              setSelectedIndex(index);
              handleChange?.(name);
```